Meadowlark-js/packages/meadowlark-core/src/validation/DocumentIdentityExtractor.ts

```diff
@@ -32,8 +32,18 @@ type NullableTopLevelEntity = { superclass: TopLevelEntity | null };
  */
 function singleIdentityFrom(property: EntityProperty, body: object, documentPath: string[]): DocumentIdentity {
   const { apiMapping } = property.data.meadowlark as EntityPropertyMeadowlarkData;
-  const documentPathAsString: string = [...documentPath, apiMapping.fullName].join('.');
-  const elementValue: string | undefined = R.path([...documentPath, apiMapping.fullName], body);
+
+  let path: string[];
+
+  if (property.type === 'schoolYearEnumeration' && property.roleName) {
+    path = [...documentPath, 'schoolYear'];
```

Comment:
I could have used `property.metaEdName` or something like that here, but chose to hard code `schoolYear` so that it is more explicit & obvious.

Without this clause, the path was coming out as `graduationSchoolYearTypeReference.graduationSchoolYear` because the `apiMapping.fullName == 'graduationSchoolYear'`.